keystore-core/src/main/java/net/spals/appbuilder/keystore/core/KeyStoreProvider.java

```diff
@@ -0,0 +1,61 @@
+package net.spals.appbuilder.keystore.core;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.netflix.governator.annotations.Configuration;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigException;
+import net.spals.appbuilder.annotations.service.AutoBindProvider;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * A {@link Provider} which provides a {@link KeyStore}.
+ *
+ * @author tkral
+ */
+@AutoBindProvider
+public class KeyStoreProvider implements Provider<KeyStore> {
+
+    /**
+     * Creates a {@link KeyStore} outside of dependency injection.
+     * <p>
+     * This can be used for use cases that fall outside of an
+     * application's dependency graph. In particular, this is useful
+     * for configuration encryption.
+     *
+     * @return A new {@link KeyStore} based on the given {@link Config}.
+     * @throws ConfigException If the given {@link Config} is not properly
+     *         configured to create a {@link KeyStore} instance.
+     */
+    public static KeyStore createKeyStore(final Config serviceConfig) {
+        final String storeSystem = serviceConfig.getString("keyStore.system");
+
+        if ("password".equals(storeSystem)) {
```

Comment:
FYI, a new java idiom I've read is java.util.Objects.equals(...) , clarifying versus the guava Objects. This is fine, just wanted to update you with your thoughts. I have it as a setting in intellij to switch to Objects.equals. It's nice to not think about NPE's.